diagnostic_feedback/public/js/src/student/student_quiz.js

```diff
@@ -141,18 +147,40 @@ function StudentQuiz(runtime, element, initData) {
         data: JSON.stringify(choice),
         success: function (response) {
           success = response.success;
-
-          if (success && response.student_result) {
-            //log event for quiz finish
-            event_data.event_type = 'xblock.diagnostic_feedback.quiz.finish';
-            common.publishEvent(event_data);
-            showResult(response.student_result);
-          }
-
-          if (success) {
+          var questionEventData = getQuestionEventData(choice.student_choice),
+            event_data = {
+              event_type: '',
+              question_txt: questionEventData.question_txt,
+              student_choice: questionEventData.student_choice,
+              current_question: currentStep,
+              is_last_question: isLast
+            };
+
+          if(success){
             //log event for question submission success
             event_data.event_type = 'xblock.diagnostic_feedback.quiz.question.submitted';
             common.publishEvent(event_data);
+
+            //log event for loading question
```

Comment:
@jareerahsan Nit: Move this comment inside the body of `if (!isLast)` to be consistent with the `else if (response.student_result)` part below.